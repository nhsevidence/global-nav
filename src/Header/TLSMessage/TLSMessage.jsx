import React, { Component } from "react";

import styles from "./TLSMessage.module.scss";

export default class TLSMessage extends Component {
	// See https://gist.github.com/padolsey/527683#gistcomment-768383
	getIEVersion() {
		let tmp = (document["documentMode"] || document.attachEvent) && "ev",
			msie =
				tmp &&
				(tmp = window[tmp + "al"]) &&
				tmp("/*@cc_on 1;@*/") &&
				+((/msie (\d+)/i.exec(navigator.userAgent) || [])[1] || 0);
		return msie || void 0;
	}

	render() {
		const ieVersion = this.getIEVersion();

		if (!ieVersion || ieVersion > 10) {
			return null;
		}

		return (
			<div className={styles.alert}>
				Your version of Internet Explorer (version {ieVersion}) doesnt support
				TLS 1.2.
			</div>
		);
	}
}
