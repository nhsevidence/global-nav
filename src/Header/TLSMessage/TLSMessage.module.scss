@import '~@nice-digital/design-system/src/stylesheets/settings/settings';
@import '~@nice-digital/design-system/src/stylesheets/helpers/helpers-utils';

.container {
  margin: 0 auto rem($spacing-medium) auto;
  max-width: rem($nice-container-max-width);
  // TODO Replace with mixin from nice design system
  width: 96%;
  width: calc(100% - #{rem($spacing-large)});
}

.alert {
  background: $colour-error-background;
  border: 1px solid $colour-error;
  color: $colour-error;
  margin: rem($spacing-medium) auto 0 auto;
  max-width: rem($nice-container-max-width);
  padding: rem($spacing-medium);

  > :first-child {
    margin-top: 0;
  }
}

.lastP {
  margin-bottom: 0;
}
