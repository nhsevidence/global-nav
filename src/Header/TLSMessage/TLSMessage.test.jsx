import React from "react";
import TLSMessage from "./TLSMessage";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("TLSMessage", () => {
	it("Renders without crashing", () => {
		const wrapper = shallow(<TLSMessage />);
		expect(wrapper).toHaveLength(1);
	});

	it("Doesn't render in IE11", () => {
		const wrapper = shallow(<TLSMessage />);

		wrapper.instance().getIEVersion = jest.fn(() => 11);
		wrapper.instance().forceUpdate();

		expect(wrapper.get(0)).toBeNull();
	});

	it("Doesn't render in modern browsers", () => {
		const wrapper = shallow(<TLSMessage />);

		wrapper.instance().getIEVersion = jest.fn(() => null);
		wrapper.instance().forceUpdate();

		expect(wrapper.get(0)).toBeNull();
	});

	it("Matches snapshot in old IE", () => {
		const wrapper = shallow(<TLSMessage />);

		wrapper.instance().getIEVersion = jest.fn(() => 8);
		wrapper.instance().forceUpdate();

		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
