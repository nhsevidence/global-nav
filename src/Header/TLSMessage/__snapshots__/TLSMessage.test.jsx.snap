// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TLSMessage Matches snapshot in old IE 1`] = `
<aside
  className="container"
>
  <div
    className="alert"
  >
    <h2>
      TLS upgrade warning - your browser is out of date
    </h2>
    <p>
      Your version of Internet Explorer (version 
      8
      ) is outdated and insecure.
    </p>
    <p>
      NICE will be upgrading websites to latest security standards (TLS 1.2) from
       
      <strong>
        <time
          dateTime="2019-08-01"
        >
          1 August 2019
        </time>
      </strong>
      .
    </p>
    <p>
      After this date, NICE websites may stop working in Internet Explorer versions 8, 9 and 10.
    </p>
    <p>
      Please upgrade your browser to continue accessing NICE services after this date.
    </p>
    <p
      className="lastP"
    >
      <a
        href="https://www.nice.org.uk/tls-upgrades"
      >
        Find out how to upgrade your browser
      </a>
    </p>
  </div>
</aside>
`;
